diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        val list = mutableListOf<ASTNode>()
+        var leafWithTypeCount = 0
+        var firstNode: ASTNode? = null
+        applyToCode(code){node ->
+            if(firstNode == null)
+                firstNode = node
+            if (node.isLeaf() && node.elementType == WHITE_SPACE) {
+                leafWithTypeCount++
+            }
+        }
+        firstNode?.getAllLLeafsWithSpecificType(WHITE_SPACE, list)
+        Assert.assertEquals(list.size, leafWithTypeCount)
+    }
+
+    @Test
+    fun `test findLeafWithSpecificType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var firstNode: ASTNode? = null
+        var resultNode: ASTNode? = null
+        applyToCode(code){node ->
+            if(firstNode == null)
+                firstNode = node
+            if (resultNode == null && node.elementType == CLASS_BODY) {
+                resultNode = node
+            }
+        }
+        firstNode = firstNode?.findLeafWithSpecificType(CLASS_BODY)
+        Assert.assertEquals(resultNode?.text, firstNode?.text)
+    }
+
+    @Test
+    fun `test findAllNodesWithSpecificType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var firstNode: ASTNode? = null
+        var listResults = mutableListOf<ASTNode>()
+        applyToCode(code){node ->
+            if(firstNode == null)
+                firstNode = node
+            if (node.elementType == IDENTIFIER) {
+                listResults.add(node)
+            }
+        }
+        val listTypes = firstNode?.findAllNodesWithSpecificType(IDENTIFIER)
+        Assert.assertEquals(listResults, listTypes)
+    }
+
+
+    @Test
+    fun `test isAccessibleOutside`(){
+        var code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                private fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.elementType == MODIFIER_LIST)
+                Assert.assertFalse(node.isAccessibleOutside())
+        }
+        code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.elementType == MODIFIER_LIST)
```

Comment:
Is this condition true for this example? I guess, you should have explicit modifier for this node to be present.